gobblin-compaction/src/main/java/gobblin/compaction/mapreduce/MRCompactorJobRunner.java

```diff
@@ -75,57 +98,73 @@
   protected final FileSystem fs;
   protected final FsPermission perm;
   protected final boolean deduplicate;
+  protected final double priority;
   protected final EventSubmitter eventSubmitter;
 
-  protected MRCompactorJobRunner(State jobProps, FileSystem fs) {
-    this.jobProps = jobProps;
-    this.topic = jobProps.getProp(ConfigurationKeys.COMPACTION_TOPIC);
-    this.inputPath = new Path(jobProps.getProp(ConfigurationKeys.COMPACTION_JOB_INPUT_DIR));
-    this.outputPath = new Path(jobProps.getProp(ConfigurationKeys.COMPACTION_JOB_DEST_DIR));
-    this.tmpPath = new Path(jobProps.getProp(ConfigurationKeys.COMPACTION_JOB_TMP_DIR));
+  private volatile Policy policy = Policy.DO_NOT_PUBLISH_DATA;
+  private volatile Status status = Status.RUNNING;
+
+  protected MRCompactorJobRunner(Dataset dataset, FileSystem fs, Double priority) {
+    this.dataset = dataset;
+    this.jobProps = dataset.jobProps();
```

Comment:
Ok. I'll wait to see for the new PR with the integrated datasets. In general, if those are associated with the dataset, they should be "datasetProps", not "jobProps". 

Further,  as an editorial comment, I think we have way too many property collections. It makes it hard to figure out what is the life-cycle of the collections and the properties, which are read/only and which are read/write, and all kinds of other data dependencies. I'd love to see more well-type java objects and interfaces and less property collections.  This will also make us think a little bit more what the API is instead of just dumping stuff into a collection.